lib/processor/archive.go

```diff
@@ -198,6 +202,39 @@ func linesArchive(hFunc headerFunc, msg types.Message) (types.Part, error) {
 	return newPart, nil
 }
 
+func concatenate(slices [][]byte) []byte {
+	if len(slices) == 0 {
+		return []byte{}
+	}
+	if len(slices) == 1 {
+		return append([]byte(nil), slices[0]...)
+	}
+	totalLen := 0
+	for _, slice := range slices {
+		totalLen += len(slice)
+	}
+
+	retr := make([]byte, totalLen, totalLen)
+	pos := 0
+	for _, slice := range slices {
+		pos += copy(retr[pos:], slice)
+	}
+	return retr
+}
+func allBytesArchive(hFunc headerFunc, msg types.Message) (types.Part, error) {
+	tmpParts := make([][]byte, msg.Len())
+	err := msg.Iter(func(i int, part types.Part) error {
+		tmpParts[i] = part.Get()
```

Comment:
```suggestion
		buf.Write(part.Get())
```

Reply:
@Jeffail
Many thanks for looking into this. My choice has todo with my mental model of the performance of  languages with a genrational garbage collector. I assume that is less costly to allocate many fixed length small arrays for a short time than to dynamically grow a single buffer. I did not benchmark though and am very new to golang.

I therefore suggest that you choose what implementation feels more idiomatic to you and matches the existing codebase.

Reply:
@Jeffail Regarding the naming `all-bytes`: I found it quite supristing that throughout benthos an empty `delimiter` option is synonymous to `'\n'`. I expected it to be what is labeled `all-bytes` here. 

You're right: `concatenate` seems a nice descriptive name for this operation.